quarkus.ls/com.redhat.quarkus.ls/src/main/java/com/redhat/quarkus/services/QuarkusCodeActions.java

```diff
@@ -107,6 +111,77 @@ private void doCodeActionsForUnknown(Diagnostic diagnostic, PropertiesModel docu
 		}
 	}
 
+	/**
+	 * Returns the <code>Position</code> to insert the missing required code action property into
+	 * @param textDocument the text document
+	 * @return the <code>Position</code> to insert the missing required code action property into
+	 * @throws BadLocationException
+	 */
+	private Position getPositionForRequiredCodeAction(TextDocument textDocument) throws BadLocationException {
+		String textDocumentText = textDocument.getText();
+
+		Position position;
+		
+		if (!StringUtils.hasText(textDocumentText)) {
+			position = new Position(0, 0);
+		} else {
+			position = textDocument.positionAt(StringUtils.trimEnd(textDocumentText).length());
+		}
+		return position;
+	}
+
+	/**
+	 * Returns the missing required property name from <code>diagnosticMessage</code>
+	 * @param diagnosticMessage the diagnostic message containing the property name in single quotes
+	 * @return the missing required property name from <code>diagnosticMessage</code>
+	 */
+	private String getPropertyNameFromRequiredMessage(String diagnosticMessage) {
+		String property = diagnosticMessage.substring(diagnosticMessage.indexOf("'") + 1);

```

Comment:
Here you create 2 String instances (because you are calling 2 substring).

I will replace with this code (not tested):

```java
private String getPropertyNameFromRequiredMessage(String diagnosticMessage) {
	int start = diagnosticMessage.indexOf('\'') + 1;
	int end = diagnosticMessage.indexOf('\'', start);
	return diagnosticMessage.substring(start, end);
}
```